swb.md

```diff
@@ -0,0 +1,46 @@
+The chakra implementation of concurrent GC and partial GC are based on background scanning the dirty pages. On windows, the OS kernel provides a feature called Write Watch( we also call it hardware write barrier), which the system keeps track of the pages that are written to in the committed memory region, ChakraCore GC use this information to determine which recycler allocated pages are dirty and do background rescan/mark on them. On non-Windows platform the hardware write barrier is not available, so we use software write barrier to track the dirty pages. In implementation of software write barrier ChakraCore uses a card table to map the whole address space, every page in the process has one byte in the card table corresponding to it. And all the recycler allocated non-Leaf struct have smart pointer wrapped to all its recycler pointer fields, while updating the pointer field, the smart pointer will update the corresponding byte in the card table, so the background thread knows the page is dirty. 
```

Comment:
Please have a title, and divide content into a few sections for easier navigate. The plugin explanation will need an anchor to directly jump to.